### DISCUSSION TITLE
Alamofire on IOS

### DISCUSSION THREAD

**Discussion by StaffyBoy** [Alamofire/Alamofire#3918] [Q&A]
I have a problem with MS SharePoint on my iPhone. Each time I try to use the app it gives me an error. It says "Something went Wrong" and it identifies the following error: Alamofire.AFError error 3. This has been happening for months. I have not managed to use this app at all.
I my communication with Microsoft Support we exchanged the following questions and answers.
                                                                                           = = = 
While I'm checking your issue, I'm going to ask some questions that can help me to provide you a resolution.
1. Network and Connectivity:
		o Are you connected to a stable Wi-Fi or cellular network? Yes, both are stable.
		o Have you tried accessing SharePoint on a different network? Yes
2. App and Device Information:
		o What version of the SharePoint app are you using? 4.53.3
		o Is your iPhone’s iOS updated to the latest version? IOS 18.01
		o Have you tried reinstalling the SharePoint app? Many times, over a period of months.
3. Permissions and Access:
		o Do you have the necessary permissions to access the SharePoint site? I am a 365 Admin.
		o Can you access the SharePoint site from a web browser on your iPhone or another device? Yes, daily.
4. Error Specifics:
		o Does the error occur with all SharePoint sites or just a specific one? All/Any on my iPhone
		o Are there any additional error messages or codes displayed? Not aware of any others. I have seen this one many times.
5. Cache and Data:
		o Have you tried clearing the cache of the SharePoint app? I believe I have. 
		o Have you recently changed any settings or configurations in the app? No, but I have fiddled with some of them to try and get this working. SP has never worked on my phone.
6. Other Devices:
		o Can you access the SharePoint site from another device, such as a computer or another mobile device? Yes
The error message you encountered, “The operation couldn’t be completed: Alamofire.AFError error 3,” indicates a problem related to Alamofire.Alamofire is a Swift-based HTTP networking library that simplifies tasks such as making network requests, handling responses, and managing network reachability on iOS devices.
Since the issue is specific to your app and does not occur on the web or other devices, it is possible that there is an issue with Alamofire.

**Comment by mc7h** [comment]
> it is possible that there is an issue with Alamofire.

No. That's very unlikely. It's more likely an issue with your app configuration, networking or Sharepoint's integration of AF.  

I'm sorry you were fobbed off by Microsoft Support.

The issue is relates to the URL parameter encoding, so check to see your Sharepoint instance's URL is set up correctly (If that's an option for you)

**Comment by StaffyBoy** [comment]
Hello Marc,
I tried SharePoint for IOS on my iPad and iPhone for about 2 years. Although I could see how potentially useful it might be this never materialised.
My conclusions:

  *   Whilst the performance on iPad was marginally better, the program never demonstrated the stability of a business tool. It’s a gimmick at best.
  *   There is no meaningful support.
  *   I was told that it may be the way in which the program was installed. However, IOS doesn’t offer any options in this regard.

SharePoint for IOS is just not a program to be taken seriously!

Regards

Paul Haitas
Iron Fireman SA (Pty) Ltd
